cpp/ReadMe.md

```diff
@@ -0,0 +1,6 @@
+
+    We provide a simple base implementation of a Record datastructure and collection of Records
+    We would like to implement FindMatchingRecords function that searches through the records for a string
+    We have provided a base FindMatchingRecords Function that we would like to optimize
+    Please provide your own implementation of Record, RecordCollection and FindMatchingRecords
+    Feel free to change the profiler function, add unit tests, move the definitions into separate classes.
```

Comment:
probably should add some things they shouldn't assume.

- don't assume QBRecordCollection is sorted by string value
- QBRecordCollection input id's are monotonically increasing.
- ???